Update Rust and crates

BlackDex opened the pull request:
- Updated Rust to v1.83.0
- Updated MSRV to v1.82.0 (Needed for html5gum crate)
- Updated icon fetching code to match new html5gum version
- Updated workflows
- Enabled edition 2024 clippy lints Nightly reports some clippy hints, but that would be too much to change in this PR i think.
- Added crate patches for the fern and yubico crates to update more dependencies
- Fixed an issue with the diesel_logger in combination with the SQLite backup feature

BlackDex commented:
Looks like the Rust v1.83.0 images are not yet available

dfunkt commented:
Perhaps we could also use commit https://github.com/daboross/fern/commit/3e775ccfafe7d24baee39826d38011981b2e55b5 as an override for fern in order to have `syslog-7` support?

BlackDex commented:
@dfunkt, yea that might be a good one.
I have updated it, and fixed an issue with the query_logger (diesel-logger) feature too.

Looks like the image is still not available :(.

BlackDex commented:
We could keep the docker files on v1.82.0, the correct version will be pulled anyway.

BlackDex commented:
@dani-garcia, we could also patch yubikey-rs which causes several duplicate dependencies to be removed.
Mainly reqwest and all it's dependencies.

It would then be based from my forked repo where the PR is coming from.
```toml
yubico = { git = "https://github.com/BlackDex/yubico-rs", rev = "00df14811f58155c0f02e3ab10f1570ed3e115c6" }
```

dfunkt commented on Cargo.toml:
```diff
@@ -166,6 +166,11 @@ rpassword = "7.3.1"
+[patch.crates-io]
+# Patch fern to support syslog v7
+fern = { git = "https://github.com/daboross/fern", rev = "3e775ccfafe7d24baee39826d38011981b2e55b5" }
+yubico = { git = "https://github.com/BlackDex/yubico-rs", rev = "00df14811f58155c0f02e3ab10f1570ed3e115c6" }
```
```suggestion

# Patch yubico to remove duplicate dependencies
yubico = { git = "https://github.com/BlackDex/yubico-rs", rev = "00df14811f58155c0f02e3ab10f1570ed3e115c6" }
```

dani-garcia commented on src/main.rs:
```diff
@@ -201,25 +201,20 @@ fn parse_args() {
-        err_silent!("The 'sqlite' feature is not enabled. Backups only works for SQLite databases")
+async fn backup_sqlite() -> Result<String, Error> {
+    use crate::db::{backup_database, DbConnType};
+    if DbConnType::from_url(&CONFIG.database_url()).map(|t| t == DbConnType::sqlite).unwrap_or(false) {
```
Do we need to check for the database type here? We're already doing that inside `backup_database` as far as I can see.

dani-garcia reviewed (approved):
Yeah, patching the dependency for now sounds good, we don't want to be dragging lots of old transitive dependencies if we can avoid it.
